src/PlatformEngines.jl

```diff
@@ -355,12 +355,12 @@ function download_verify(
             # Download and verify from scratch
             download(url, dest; verbose=verbose || !quiet_download)
             if hash !== nothing && !verify(dest, hash; verbose=verbose)
-                error("Verification failed")
+                error("Verification failed. Download does not match expected hash")
```

Comment:
Might as well print the computed and expected hash here?

Reply:
It's already logged, but yeah that could be fed through as an error

Reply:
Okay if logged perhaps it's OK.

Reply:
I did want to make it more streamlined that but the `@error` is deep in PlatformEngines, and I couldn't figure out a tidy way to do it. Perhaps for another PR